### PR TITLE
add chghost capability support

### PR DESCRIPTION
**Pull request by ailin-nemui**

### CODE REVIEW

File: src/fe-common/core/fe-messages.c [hunk 1]
```diff
@@ -359,6 +359,77 @@ static void sig_message_join(SERVER_REC *server, const char *channel,
...
+		if (once)
+			g_string_append_printf(chans, "%s,", rec->visible_name);
+		else {
+			window = window_item_window((WI_ITEM_REC *) rec);
+			if (g_slist_find(windows, window) == NULL) {
+				windows = g_slist_append(windows, window);
+				printformat(server, rec->visible_name,
+					    level,
+					    TXT_HOST_CHANGED, nick, address, old_address,
+					    rec->visible_name);
```
[the remark this answers is not in the document]
Comment (ailin-nemui) [reply src/fe-common/core/fe-messages.c]: I don't know, originally the host change message causes a quit AND join message. any other @irssi/developers with an opinion?

[the remark this answers is not in the document]
Comment (dequis) [reply src/fe-common/core/fe-messages.c]: I think it's quit/join on freenode, but absolutely nothing in other networks. Yeah, could use other opinions.

File: src/fe-common/core/fe-messages.c [hunk 2]
```diff
@@ -360,89 +360,125 @@ static void sig_message_join(SERVER_REC *server, const char *channel,
...
-
-	ignore_check_plus(server, nick, address, channel, NULL, &level, FALSE);
-
-	printformat(server, channel, level,
-		    TXT_PART, nick, address, channel, reason);
-}
-
-static void sig_message_quit(SERVER_REC *server, const char *nick,
-			     const char *address, const char *reason)
+static void spread_server_message_to_windows(SERVER_REC *server, gboolean once,
```
Comment (dequis) [inline src/fe-common/core/fe-messages.c]: nice

File: src/fe-common/core/fe-messages.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ailin-nemui) [reply src/fe-common/core/fe-messages.c]: also on quakenet and undernet

**Pull request merged**